lib/marionette/drivers/tcp-sync.js

```diff
@@ -60,6 +60,10 @@ TcpSync.prototype.waitForSocket = function(options, callback) {
   var start = Date.now();
   var self = this;
 
+  // Use sockittome's built in polling timeout during calls to connect
+  // to avoid socket misuse.
+  sockit.setPollTimeout(interval);
```

Comment:
Interesting. Should we do what we do in the constructor and use SOCKET_TIMEOUT_EXTRA?

https://github.com/mozilla-b2g/marionette-js-client/blob/25d1b2d4c083739bb0cd0648b548c9d153ef5859/lib/marionette/drivers/tcp-sync.js#L24